test/pom.xml

```diff
@@ -182,6 +182,12 @@ THE SOFTWARE.
       <version>4.0</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>cloudbees-folder</artifactId>
+      <version>6.0.4</version>
```

Comment:
Folders will depend on a different version of `jenkins-core` which will have different transitive dependency versions than those from `jenkins-core" in the reactor. 

As the `cloudbees-folder` dependency is "closer" it can win for some of the conflicts. That means that the tests will not be running with the same dependencies that we bundle in the .war.

It would be different if we were using the plugin manager to load the hpi, but instead we let Maven Surefire setup the classpath (with test classes first) so that the plugin classes are visible from core.

I think you need to find a different way to test